### PR TITLE
[azservicebus] Vendoring go-amqp

### PR DESCRIPTION
**Pull request by richardpark-msft**
There are enough interface changes for go-amqp that we want to insulate this package somewhat. Simplest solution that we could come up with was to temporarily vendor go-amqp here. 

So this PR is a literal copy of go-amqp's code after I did a `go mod vendor` with:
- Fixing up of references to use our internally vendored copy of go-amqp:
- Adding in a NOTICE.txt to make sure we credit Kale and respect the sub-license.

### CODE REVIEW

File: sdk/messaging/azservicebus/NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+           azgosdkhelp@microsoft.com
+
+The attached notices are provided for information only.
+
+License notice for go-amqp
+------------------------------------------------------------------------------
+
+    MIT License
+
+    Copyright (C) 2017 Kale Blankenship
```
Comment (richardpark-msft) [inline sdk/messaging/azservicebus/NOTICE.txt:36]: @RickWinter, this is an exact copy of what's in the LICENSE.txt for go-amqp.

File: sdk/messaging/azservicebus/internal/go-amqp/.gitattributes [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Binary files (no line-ending conversions), diff using hexdump
```
Comment (jhendrixMSFT) [inline sdk/messaging/azservicebus/internal/go-amqp/.gitattributes:1]: Do we need this file?  Looks like a few others could be pruned too.

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/internal/go-amqp/.gitattributes:1]: Yeah, let me go look. I tried to remove a few obvious ones.

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/internal/go-amqp/.gitattributes]: Got rid of some more files.

File: sdk/messaging/azservicebus/internal/go-amqp/client.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
+// Copyright (C) 2017 Kale Blankenship
```
Comment (richardpark-msft) [inline sdk/messaging/azservicebus/internal/go-amqp/client.go:1]: @RickWinter  - an example of our "combined" copyright (obviously subject to whatever we need to change).

File: sdk/messaging/azservicebus/internal/go-amqp/doc.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+AMQP 1.0 is not compatible with AMQP 0-9-1 or 0-10, which are
+the most common AMQP protocols in use today.
+
+The example below shows how to use this package to connect
+to a Microsoft Azure Service Bus queue.
+*/
+
+// Copyright (C) 2017 Kale Blankenship
+// Portions Copyright (c) Microsoft Corporation
+package amqp // import "github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal/go-amqp"
```
Comment (jhendrixMSFT) [inline sdk/messaging/azservicebus/internal/go-amqp/doc.go:13]: Trailing import comment?

**Review by RickWinter — APPROVED** [review]

File: sdk/messaging/azservicebus/internal/go-amqp/doc.go (hunk 4 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/internal/go-amqp/doc.go]: Removed.

**Pull request merged**